### DISCUSSION TITLE
simple-websocket and gunicorn

### DISCUSSION THREAD

**Discussion by maybephilipp** [miguelgrinberg/python-engineio#335] [Q&A] [answered]
Hello everyone!

Could someone please explain why gunicorn is excluded from support of `simple-websocket`? I see when `simple-websocket` support was implemented, gunicorn was restricted: [this commit](https://github.com/miguelgrinberg/python-engineio/commit/614f564275c635dc8b03d33dab44bf80d280cbcc) `src/engineio/async_drivers/_websocket_wsgi.py`, line 16-17

1. Why?
2. Could we plan adding a note about that in docs, and, if applicable, change docs regarding usage with Gevent and Gunicorn: https://python-socketio.readthedocs.io/en/stable/server.html#gevent

Thank you for your work!

**Comment by miguelgrinberg** [comment, marked as answer]
I don't understand this. Gunicorn works fine with simple-websocket. The threaded, eventlet and gevent workers in Gunicorn all should work. What did you see in that commit that suggests this is not implemented? If you refer to the `StopIteration` exception, that is actually required by Gunicorn, it is not intended to prevent Gunicorn from being used, but to ensure that it works well.

**Reply by maybephilipp** [reply]
I feel stupid, thank you for explanation! I didn't know that gunicorn requires this trick

**Reply by maybephilipp** [reply]
And regarding docs update, should I raise this question in SocketIO repository? Or create an issue?

Currently the documentation still states that for websocket support with Gevent, I need to use gevent-websocket. Same for Gevent + Gunicorn.

https://python-socketio.readthedocs.io/en/stable/server.html#gevent

**Reply by miguelgrinberg** [reply]
Yes, please write an issue (not a discussion) so that I don't forget to update the docs. Thanks.
